### PR TITLE
[Keyboard] Concertina

### PR DESCRIPTION
**Pull request by veikman**
A novel handwired keyboard.

## Description

For a general description with photos, see https://viktor.eikman.se/article/the-concertina/.

It’s the physical shape that’s new. No major customizations have been made to the firmware.

I would like to provide Configurator support, but because the shape is highly three-dimensional, **`info.json` is incomplete** and self-contradictory in this PR. I have found [this tool](https://qmk.fm/converter/) and [this guide](https://git.intruder.network/guilherme/qmk_firmware/commit/d8f090e5592fd5b6fde35ae36e36f2473c716312) but a meaningful `info.json` would seem to require getting a local instance of Configurator up and running just to see how the visualization would eventually be rendered as I hand-craft the offsets. If I should I be doing that, please let me know.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes **except** `info.json`, which is incomplete.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by veikman** [comment]
The second commit (passing lint checks above) has a stubbed-out `info.json` that is probably going to look bad and be useless, partly because it is not clear to me how the order of the JSON `layout` maps to the C macro, partly because I have not previewed how Configurator will render it.

File: keyboards/handwired/concertina/64key/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,117 @@
+#include "64key.h"
```
Comment (drashna) [inline keyboards/handwired/concertina/64key/keymaps/default/keymap.c:1]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/handwired/concertina/concertina.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/concertina/concertina.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/concertina/config.h [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/concertina/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/concertina/64key/64key.h [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+           RT_12, RT_22, RT_32,               \
+    RT_01, RT_11, RT_21, RT_31,               \
+    RT_00, RT_10, RT_20,                      \
+                                              \
+    RM_04, RM_14, RM_24, RM_34,               \
+    RM_03, RM_13, RM_23, RM_33, RM_43, RM_53, \
+    RM_02, RM_12, RM_22, RM_32, RM_42, RM_52, \
+           RM_11, RM_21, RM_31, RM_41, RM_51, \
+                  RM_20                       \
+  ) \
```
Comment (drashna) [inline keyboards/handwired/concertina/64key/64key.h:43]:
```suggestion
#define LAYOUT_64key( \
                  LM_34, LM_24, LM_14, LM_04,     RM_04, RM_14, RM_24, RM_34,               \
    LM_53, LM_43, LM_33, LM_23, LM_13, LM_03,     RM_03, RM_13, RM_23, RM_33, RM_43, RM_53, \
    LM_52, LM_42, LM_32, LM_22, LM_12, LM_02,     RM_02, RM_12, RM_22, RM_32, RM_42, RM_52, \
    LM_51, LM_41, LM_31, LM_21, LM_11,                   RM_11, RM_21, RM_31, RM_41, RM_51, \
                         LM_20,                                 RM_20                       \
                  LT_32, LT_22, LT_12,                   RT_12, RT_22, RT_32,               \
                  LT_31, LT_21, LT_11, LT_01,     RT_01, RT_11, RT_21, RT_31,               \
                         LT_20, LT_10, LT_00,     RT_00, RT_10, RT_20,                      \
  ) \
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/concertina/64key/64key.c [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#include "64key.h"
```
[the remark this answers is not in the document]
Comment (veikman) [reply keyboards/handwired/concertina/64key/64key.c]: Done in third commit.

File: keyboards/handwired/concertina/64key/64key.c [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+
+void matrix_init_kb(void) {
+	matrix_init_user();
+};
```
[the remark this answers is not in the document]
Comment (veikman) [reply keyboards/handwired/concertina/64key/64key.c]: Done in third commit.

File: keyboards/handwired/concertina/64key/64key.h [hunk 7]
```diff
@@ -0,0 +1,53 @@
+#pragma once
```
[the remark this answers is not in the document]
Comment (veikman) [reply keyboards/handwired/concertina/64key/64key.h:17]: Done in third commit.

File: keyboards/handwired/concertina/64key/keymaps/default/config.h [hunk 8]
```diff
@@ -0,0 +1,7 @@
+#pragma once
```
[the remark this answers is not in the document]
Comment (veikman) [reply keyboards/handwired/concertina/64key/keymaps/default/config.h:17]: Done in third commit.

File: keyboards/handwired/concertina/64key/keymaps/default/keymap.c (hunk 1 above)
Comment (veikman) [reply keyboards/handwired/concertina/64key/keymaps/default/keymap.c]: Done in third commit.

File: keyboards/handwired/concertina/concertina.h (hunk 2 above)
Comment (veikman) [reply keyboards/handwired/concertina/concertina.h:17]: Done in third commit.

File: keyboards/handwired/concertina/config.h (hunk 3 above)
Comment (veikman) [reply keyboards/handwired/concertina/config.h:17]: Done in third commit.

File: keyboards/handwired/concertina/64key/64key.h (hunk 4 above)
Comment (veikman) [reply keyboards/handwired/concertina/64key/64key.h]: Done in third commit. Keymap altered to match the new matrix. If there is a related policy document, I missed it. If not, there should be.

File: keyboards/handwired/concertina/64key/config.h [hunk 9]
```diff
@@ -0,0 +1,18 @@
+#pragma once
```
[the remark this answers is not in the document]
Comment (veikman) [reply keyboards/handwired/concertina/64key/config.h:17]: Done in third commit.

**Review by drashna — APPROVED** [review]

**Review by spidey3 — APPROVED** [review]

**Comment by veikman** [comment]
Thank you for the review and merge, but I would still be interested to know how to preview Configurator layouts before submitting a PR. Right now, Configurator shows [no known layouts](https://config.qmk.fm/#/handwired/concertina/to_be_defined) for the board. I guess I have failed somehow, but I don’t know where to start looking to fix this.

**Comment by noroadsleft** [comment]
@veikman,

> Thank you for the review and merge, but I would still be interested to know how to preview Configurator layouts before submitting a PR.

You can preview `info.json` files by opening QMK Configurator in your web browser and hitting <kbd><kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd></kbd>, in that order.

> Right now, Configurator shows [no known layouts](https://config.qmk.fm/#/handwired/concertina/to_be_defined) for the board. I guess I have failed somehow, but I don’t know where to start looking to fix this.

Your submitted directory structure is wrong. You have source split between the `handwired/concertina` and `handwired/concertina/64key` directories. This isn't a problem on its own, but the `64key` directory has no `rules.mk` file, the existence of which is required for QMK's Makefile to understand that this is a keyboard that can be compiled. You have a `rules.mk` file in `handwired/concertina`, but no `keymaps` folder therein, so `make handwired/concertina` fails because it can't find your layout macro `LAYOUT_64key` or any keymaps.

This block

https://github.com/qmk/qmk_firmware/blob/11146ecd086c6b6736ab3c5947f26a96be642671/keyboards/handwired/concertina/concertina.h#L19-L21

is never `true` because there is no `keyboards/handwired/concertina/64key/rules.mk` file, so `KEYBOARD_handwired_concertina_64key` is never defined.


I've posted a compilation fix at #11987. Would appreciate your attention there regarding the Configurator implementation, which I'm not sure is correct at this time.

**Pull request merged**